### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this Pull Request, I:

- [x] Create `Classroom` class.
- [x] Create a `has-many` relationship between `Classroom` and `Student` classes.
- [x] Create `Book` class.
- [x] Create `Rental` class.
- [x] Create a `many-to-many` relationship between `Person` , `Book` and `Rental` classes.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def initialize(age, parent_permission: true, name: 'unknown')
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rents = []
   end
```
Comment (SajjadAhmad14) [inline person.rb:19]: Please add a method for adding rentals in the `Person` class. You may see [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) lesson for reference.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class Book
+  attr_reader :rents
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rents = []
+  end
```
Comment (SajjadAhmad14) [inline book.rb:11]: Please add a method for adding rentals in the `Book` class. You may see [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) lesson for reference.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Karangwa, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by descholar-ceo — APPROVED** [review]
Hi @Ntwali-Josue , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**